apps/root/src/pages/transfer/components/recipient-address/components/contact-list-modal/index.tsx

```diff
@@ -67,65 +80,80 @@ const ContactListModal = ({ shouldShow, setShouldShow, onClickContact }: Contact
           contact.address.toLowerCase().includes(searchValue.toLowerCase()) ||
           contact.label?.toLowerCase().includes(searchValue.toLowerCase())
       ),
-    [contactList, searchValue]
+    [contactList, searchValue, updateTrigger]
   );
 
+  const goBackToTransfer = () => {
+    setOpen(false);
+    setOpenAddContactModal(false);
+  };
+
   return (
-    <Modal
-      open={shouldShow}
-      onClose={() => setShouldShow(false)}
-      closeOnBackdrop={true}
-      title={<FormattedMessage description="contactList" defaultMessage="Contact list" />}
-      headerButton={
-        contactList.length !== 0 && (
-          <Button variant="outlined" color="primary" onClick={onOpenAddContactForm}>
-            <Typography variant="body1" fontWeight="bold">
+    <>
+      <AddContactModal
```

Comment:
To prevent this issue, I added the following condition to the larger 'ContactList' Modal:
```open={open && !openAddContactModal}```
I attempted to merge both modals into a single component, but encountered complications with prop drilling, as I was trying to maintain separation of concerns by keeping everything in distinct components. If you believe this approach isn't completely robust, I'm open to reviewing it further 😄